### PR TITLE
[FIX] remove base. from id of data file, add post_init_hook

### PR DESCRIPTION
**Pull request by kittiu**
Fix data file + add post_init_hook for noupdate = true to reduce following update time.

### CODE REVIEW

**Comment by kittiu** [comment]
@sbidoul for this Thai modules, should I do the merge on this repo myself? Or waiting someone to review first?

**Comment by sbidoul** [comment]
@kittiu ideally you should find reviewers and build a community. But it is understandable that you merge your own work at the beginning. That said I may have a few comments on this PR.

File: l10n_th_address/__init__.py
Comment (sbidoul) [inline l10n_th_address/__init__.py]: One may imagine situations where res.country.province is declared and populated in another module. This may happen when you install several localization at once. In this case, this update would modify data that belong to other modules. I think you can add `where module= ` to limit to your own module data.

File: l10n_th_address/__manifest__.py
Comment (sbidoul) [inline l10n_th_address/__manifest__.py]: remove the # vim line

**Review by sbidoul — CHANGES_REQUESTED** [review]
The record id [here](https://github.com/OCA/l10n-thailand/blob/802cec98b4a17a793ba0e91b8c1c4025364f1695/l10n_th_address/data/address_format.xml#L5) looks suspicious.

**Comment by sbidoul** [comment]
Is it necessary to depend on `sales_team`?

**Comment by kittiu** [comment]
I depend on `sale_team` as it has the menu Localizations, and the main propose of these Thai addresses data are for customer (for invoice addresses), so I think it is ok.

File: l10n_th_address/__init__.py
Comment (dreispt) [inline l10n_th_address/__init__.py]: Not a blocking issue, but it is best to place these init hooks in their won file, rather than in an `__init__.py`.

File: l10n_th_address/__init__.py
Comment (dreispt) [inline l10n_th_address/__init__.py]: I suggest adding `and noupdate = false`

File: l10n_th_address/__manifest__.py
Comment (dreispt) [inline l10n_th_address/__manifest__.py]: Please add the OCA to the `author` key (check the OCA module template in OCA/maintainer-tools).

Comment (sbidoul) [reply l10n_th_address/__manifest__.py]: Travis [did warn about that](https://travis-ci.org/OCA/l10n-thailand/jobs/228809869#L509) but suceeded nevertheless. That is not normal.

Comment (dreispt) [reply l10n_th_address/__manifest__.py]: I also wondered about that...

Comment (sbidoul) [reply l10n_th_address/__manifest__.py]: Also, in the website key, please use a proper URI (with http:// or https://).

Comment (dreispt) [reply l10n_th_address/__manifest__.py]: OTOH there are quite a few important MQT checks that are still "beta"/informative. This could be the case.

Comment (pedrobaeza) [reply l10n_th_address/__manifest__.py]: This is for avoiding to use our MQT in projects outside OCA

Comment (sbidoul) [reply l10n_th_address/__manifest__.py]: @pedrobaeza you mean "allowing to use..."?

Still, the check for author and license are configurable in pylint-odoo, so in our OCA configuration, these should be blocking errors no?

Comment (pedrobaeza) [reply l10n_th_address/__manifest__.py]: Sorry, I have mixed words. Yeah, that's it, for allowing to use outside OCA. Indeed, this is configurable, not at pylint-odoo level, but at https://github.com/OCA/maintainer-quality-tools. If we change the check there, anyone download that repository will find that error. Maybe we can check the target of the PR (if it's OCA/xxxx) and make it mandatory there, but I doubt it can be possible. @moylop260, any thought on this?

Comment (sbidoul) [reply l10n_th_address/__manifest__.py]: @pedrobaeza @moylop260 author check is definitely in [pylint-odoo](https://github.com/OCA/maintainer-quality-tools/blob/b76f2bc02b45e4f4fc4e386015a4422b9c71c8a8/travis/cfg/travis_run_pylint_pr.cfg#L11-L15).

So why is it not an error?

Comment (pedrobaeza) [reply l10n_th_address/__manifest__.py]: The check is there, but the activation is in MQT as said.

Comment (sbidoul) [reply l10n_th_address/__manifest__.py]: Ok, I'll open an issue in MQT to discuss that.

File: l10n_th_address/__init__.py
Comment (kittiu) [reply l10n_th_address/__init__.py]: @dreispt really not sure what you mean by won file. Beside, from what I see from other OCA addons, most if not all are doing in __init__.py, isn't it? Please advice.

File: l10n_th_address/__manifest__.py
Comment (kittiu) [reply l10n_th_address/__manifest__.py]: Fixed from the comment (so far).

File: l10n_th_address/__init__.py
Comment (dreispt) [reply l10n_th_address/__init__.py]: Sorry, I mean in "own" / separate file.
See more details at https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#installation-hooks

**Comment by kittiu** [comment]
@sbidoul your request for change https://github.com/OCA/l10n-thailand/pull/3#pullrequestreview-36314494 is done. Beside, I have moved the address_format change for base.th to post_init_hook instead coz it was noupdate="1". Is this ok?

**Review by dreispt — APPROVED** [review]

File: l10n_th_address/__manifest__.py
Comment (moylop260) [reply l10n_th_address/__manifest__.py]: The beta error is emitted:
 - `l10n_th_address/__manifest__.py:4: [C8101(manifest-required-author), ] Missing author required "Odoo Community Association (OCA)" in manifest file`:
   - https://travis-ci.org/OCA/l10n-thailand/jobs/228809869#L514

But we have these checks as beta yet.
I'll remove it from beta to affect the build result.

**Comment by moylop260** [comment]
@kittiu 
FYI we have the following beta messages (as green result):
```txt
************* Module l10n_th_address
l10n_th_address/__init__.py:1: [C7902(missing-readme), ] Missing ./README.rst file. Template here: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
l10n_th_address/views/res_partner_view.xml:6: [W7940(dangerous-view-replace-wo-priority), ]  Dangerous use of "replace" from view with priority 0 < 99
************* Module l10n_th_address.__manifest__
l10n_th_address/__manifest__.py:4: [C8103(manifest-deprecated-key), ] Deprecated key "description" in manifest file
```

https://travis-ci.org/OCA/l10n-thailand/jobs/229833197#L509-L513

Could you fix them, please?

**Comment by kittiu** [comment]
All fixed and pass all test.

**Comment by kittiu** [comment]
@sbidoul this one need more review?

**Review by sbidoul — APPROVED** [review]

**Comment by sbidoul** [comment]
@kittiu it's good for me. I recommend you squash your commits before merging to create a meaningful history.

**Comment by kittiu** [comment]
Thank you everyone. :)

**Comment by sbidoul** [comment]
Ah one last thing, when you work on pull requests, first do a fork of this repository to create the branches on your fork, so you do not pollute this repo with too many branches.

Thanks and welcome to OCA :smile:

**Comment by kittiu** [comment]
@sbidoul sure. And thank you so much for your encouragement and support.

**Pull request merged**
